collections/ansible_collections/cloudkit/service/roles/manage_finalizer/tasks/main.yaml

```diff
@@ -0,0 +1,19 @@
+- name: Get current ClusterOrder state
+  ansible.builtin.set_fact:
+    manage_finalizer_cluster_order: >-
+      {{
+        lookup('kubernetes.core.k8s', api_version=cluster_order.apiVersion, kind=cluster_order.kind,
+          resource_name=cluster_order.metadata.name, namespace=cluster_order.metadata.namespace)
+      }}
+
+- name: Add finalizer to ClusterOrder
+  when: manage_finalizer_state|default('') == "present"
```

Comment:
random question; is the default considered a good thing to have, even if the parameter is marked as `required` in the argument_specs?